src/Init/Data/Int/DivMod/Lemmas.lean

```diff
@@ -2354,6 +2354,81 @@ theorem bmod_neg_bmod : bmod (-(bmod x n)) n = bmod (-x) n := by
   apply (bmod_add_cancel_right x).mp
   rw [Int.add_left_neg, ← add_bmod_bmod, Int.add_left_neg]
 
+/- ### udiv, ediv -/
+
+theorem ediv_lt_self_of_two_le_of_zero_lt (x y : Int) (hy' : 2 ≤ y) (hx' : 0 < x):
+    x / y < x := by
+  rw [Int.div_def]
+  unfold Int.ediv
+  obtain ⟨xn, hx⟩ := Int.eq_ofNat_of_zero_le (a := x) (by omega)
+  obtain ⟨yn, hy⟩ := Int.eq_ofNat_of_zero_le (a := y) (by omega)
+  simp only [hx, hy, Int.ofNat_eq_coe, Int.ofNat_lt]
+  apply Nat.div_lt_self (by omega) (by omega)
+
+-- note that x < y → x / y = 0
+theorem zero_le_udiv_of_two_le_zero_lt (x y : Int) (hy' : 2 ≤ y) (hx' : 0 < x):
+    0 ≤ x / y := by
```

Comment:
```suggestion
theorem zero_le_ediv_of_two_le_zero_lt (x y : Int) (hy' : 2 ≤ y) (hx' : 0 < x):
    0 ≤ x / y := by
```

the same below

Reply:
done! all nomes should be consistent with this now, and I tried to relax hypotheses as much as possible